ml-agents/mlagents/trainers/sac/optimizer_torch.py

```diff
@@ -143,24 +137,32 @@ def __init__(self, policy: TorchPolicy, trainer_params: TrainerSettings):
             self.policy.actor_critic.critic, self.target_network, 1.0
         )
 
+        _total_act_size = 0
+        if self._action_spec.continuous_size > 0:
+            _total_act_size += self._action_spec.continuous_size
+        _total_act_size += sum(self._action_spec.discrete_branches)
```

Comment:
This can be accomplished with just `self._action_spec.continuous_size + sum(self._action_spec.discrete_branches)`